### DISCUSSION TITLE
Timeouts

### DISCUSSION THREAD

**Discussion by dosandco** [directus/directus#6964] [Archive]
This is my first post here - just wanted to start by saying how great I think Directus is - I'm so excited to watch this develop further.

I've been using 9 (now on rc.85) for a while now, but I noticed that in the last couple of updates, it has started to time out more and more on the API, and now I am getting the following message when using the interface directly, just navigating around:

```json
{
  "errors": [
    {
      "message": "Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

I'm installed on a Droplet with DigitalOcean, and have scaled it so that I don't appear to be using more than 15% of the resources, but that doesn't seem to have made any difference.  Is anyone else noticing this?

**Comment by rijkvanzanten** [comment]
What database are you using? Is there a specific sequence of steps that causes this behavior?

**Reply by dosandco** [reply]
It's using the Postgresql from the Digitalocean marketplace - when I do a /server/health call, I'm getting:

![image](https://user-images.githubusercontent.com/48129960/127124084-eb472341-90a5-4182-beb1-a26a2c8d0b03.png)

What does this relate to?

The sequence of steps is login > try and open any record (or even some of the documentation)

**Reply by rijkvanzanten** [reply]
Object store is the file storage. This warning says that it took longer than expected to save/retrieve a file from the storage adapter

**Reply by dosandco** [reply]
Thanks @rijkvanzanten - we have a lot of O2M links - I think we may be experiencing #6184 - we don't have _any_ files in storage - it's only the database tables itself, but some of the tables have ~100k records - we'll keep an eye out on #6184; once we are a bit more stable on our schema, caching might help

**Reply by rijkvanzanten** [reply]
@dosandco That could be, though that should be unrelated to this particular warning. 

The storage checker checks how long it takes to create, read, and then delete a near-empty file in the configured storage adapter:

https://github.com/directus/directus/blob/d1b851e34abc4fb36993f4c1b37a98aab8162b94/api/src/services/server.ts#L270-L310

In this case, it seems like your local storage took 20 seconds to create, read, and then delete the file. The server service's health checker considers that a warning, as it's above the 750ms threshold
